### PR TITLE
#802 Secure controller by avoiding priv esc

### LINKED ISSUE #802: Secure the Controller by avoiding privilege escalation issue
**Which component**:
Controller

**Is your feature request related to a problem? Please describe.**
There are some Privilege Escalation issues in the Controller
See vulnerability IDs in the vulnerability database : https://avd.aquasec.com
KSV001, KS020, KS021

**Describe the solution you'd like**
You should add these statements in the securityContext :
allowPrivilegeEscalation: false
runAsUser: UID
runAsGroup: GID

Here the UID and GID assigned should be greater than 10 000

**Describe alternatives you've considered**
No Alternative solution.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by IoannisMatzaris**
**Description of the change**
- Change the uid in the docker image
- Change the defaults for runAsUser & runAsGroup to 10001
- Change the allowPrivilegeEscalation to false
- Update the helm chart

**Benefits**
- https://avd.aquasec.com/misconfig/kubernetes/ksv001/
- https://avd.aquasec.com/misconfig/kubernetes/ksv020/
- https://avd.aquasec.com/misconfig/kubernetes/ksv021/

**Possible drawbacks**

**Applicable issues**

- fixes #802

**Additional information**

### CODE REVIEW

**Comment by alvneiayu** [comment]
thanks @IoannisMatzaris 

We are investigating your change. I will come back asap.

Álvaro

File: controller-norbac.jsonnet [hunk 1]
```diff
@@ -54,7 +54,9 @@ local namespace = 'kube-system';
               securityContext+: {
                 readOnlyRootFilesystem: true,
                 runAsNonRoot: true,
-                runAsUser: 1001,
+                allowPrivilegeEscalation: false,
+                runAsUser: 10001,
```
Comment (alvneiayu) [inline controller-norbac.jsonnet:58]: question: this 10001 user still belongs to root group?

**Review by alvneiayu — CHANGES_REQUESTED** [review]
hi @IoannisMatzaris 

First of all, thanks a lot for your time and your PR. The PR has some conflicts and I included a question and a comment.

Thanks a lot again

Álvaro

**Comment by R011y** [comment]
Did this change get merged after all? Currently trying to determine if users can safely set allowPrivilegeEscalation: false for sealed-secrets controller manifests.

**Comment by R011y** [comment]
Any update here @agarcia-oss ?? Would be good to know if the controller.yaml can support this now.

**Comment by alvneiayu** [comment]
hi @R011y 

We requested some information and changes to the user and we didn't receive answer from them. This is not already included.
We are open to continue working on this, do you want to help us contributing in this PR?

Thanks a lot

Álvaro

**Comment by alvneiayu** [comment]
If you are using the generated yaml that we are releasing, we have included: https://github.com/bitnami-labs/sealed-secrets/pull/1261. Maybe it solves your problems

**Pull request closed without merging**